pkg/reconciler/pipeline/watcher.go

```diff
@@ -11,9 +11,12 @@ import (
 // getVertexStatus will calculate the status of the vertices and return the status and reason
 func getVertexStatus(vertices *dfv1.VertexList) (bool, string) {
 	for _, vertex := range vertices.Items {
-		if !vertex.Status.IsHealthy() {
+		if vertex.Status.ObservedGeneration == 0 || vertex.Generation > vertex.Status.ObservedGeneration {
 			return false, "Progressing"
 		}
+		if !vertex.Status.IsHealthy() {
+			return false, "Unavailable"
+		}
 	}
 	return true, "Successful"
```

Comment:
sorry - if we're using "Healthy" in the other places, should we use that here too for consistency instead of "Successful"?